### PR TITLE
Ignore-expired-csrf-cookie-if-valid-auth-cookie-at-oauthcallback

### PR DESCRIPTION
**Pull request by seilersTNG**
Detailed description of the problem and the solution is described here: https://github.com/oauth2-proxy/oauth2-proxy/issues/1888 ; 

## How Has This Been Tested?

The changes have been tested by rolling out the patched version in our k8s test environment. We use keycloak-19.0.1 as auth provider.

## Checklist:

- [ ] My change requires a change to the documentation or CHANGELOG.
- [ ] I have updated the documentation/CHANGELOG accordingly.
- [ ] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

File: oauthproxy.go [hunk 1]
```diff
@@ -319,7 +319,7 @@ func (p *OAuthProxy) buildProxySubrouter(s *mux.Router) {
 	s.Path(signInPath).HandlerFunc(p.SignIn)
 	s.Path(signOutPath).HandlerFunc(p.SignOut)
 	s.Path(oauthStartPath).HandlerFunc(p.OAuthStart)
-	s.Path(oauthCallbackPath).HandlerFunc(p.OAuthCallback)
+	s.Path(oauthCallbackPath).Handler(p.sessionChain.ThenFunc(p.OAuthCallback))
```
Comment (JoelSpeed) [inline oauthproxy.go:322]: This is quite a major change to the way the project works. Previously we had always made sure that existing sessions didn't affect the attempts to re-authenticate, which seems like a valid use case to me.

What are the implications of this change?

**Review by JoelSpeed — COMMENTED** [review]
I think this change prevents a user from re-authenticating when they have an existing session, which I don't think is desirable.

I think perhaps we need to solve this in a slightly different way.

Rather than skipping trying to re-authenticate, perhaps you can recover and say, ok, we failed to re-auth, but we still have a valid session, so don't clear that?

**Comment by seilersTNG** [comment]
> This pull request has been inactive for 60 days. If the pull request is still relevant please comment to re-activate the pull request. If no action is taken within 7 days, the pull request will be marked closed.

PR is still relevant, please do not close.

**Comment by JoelSpeed** [comment]
I've removed the stale check but it looks like the last thing going on was some feedback from me, feel free to reopen if you think you'd like to continue that thread

**Pull request closed without merging**